app/src/main/java/ch/epfl/polychef/pages/LoginPage.java

```diff
@@ -68,8 +70,25 @@ public FirebaseUser getUser() {
         return FirebaseAuth.getInstance().getCurrentUser();
     }
 
+    public void prepareNextActivity(){
+        getUserStorage().initializeUserFromAuthenticatedUser(this);
+    }
+
+    @Override
+    public void onFailure() {
+        //TODO fail gracefully
+    }
+
+    @Override
+    public void onSuccess(User data) {
+        startNextActivity();
```

Comment:
Could we delete method startNextActivity and just call startActivity(new Intent(this, HomePage.class)) ?

Reply:
Yes, my bad